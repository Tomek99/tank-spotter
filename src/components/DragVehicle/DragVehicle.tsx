import React, { useState } from "react";
import Artillery from "../Vehicles/Artillery/Artillery";
import HeavyTank from "../Vehicles/HeavyTank/HeavyTank";
import MediumTank from "../Vehicles/MediumTank/MediumTank";
import Scout from "../Vehicles/Scout/Scout";
import TankDestroyer from "../Vehicles/TankDestroyer/TankDestroyer";
import { useDrag } from "@use-gesture/react";
import styles from "./DragVehicle.module.scss";

interface Vehicle {
  item: {
    id: number;
    x: number;
    y: number;
    sort: number;
  };
  settleMark: (id: number) => void;
  updateCooridnates: (id: number, x: number, y: number) => void;
}

interface Coordinate {
  x: number;
  y: number;
}

function DragVehicle({ item, settleMark, updateCooridnates }: Vehicle) {
  const [logoPos, setLogoPos] = useState<Coordinate>({ x: 0, y: 0 });

  const bindLogoPos = useDrag((params) => {
    setLogoPos({
      x: params.offset[0],
      y: params.offset[1],
    });

    updateCooridnates(item.id, params.offset[0], params.offset[1]);
  });
  return (
    <div
      {...bindLogoPos()}
      style={{
        position: "absolute",
        top: logoPos.y,
        left: logoPos.x,
        touchAction: "none",
      }}
      className={styles.DragVehicle}
      onClick={() => settleMark(item.id)}
    >
      {(() => {
        switch (item.sort) {
          case 0:
            return <Artillery />;
          case 1:
            return <HeavyTank />;
          case 2:
            return <MediumTank />;
          case 3:
            return <Scout />;
          case 4:
            return <TankDestroyer />;
          default:
            return null;
        }
      })()}
    </div>
  );
}

export default DragVehicle;
