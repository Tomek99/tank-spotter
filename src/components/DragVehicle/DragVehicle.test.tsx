import { render } from "@testing-library/react";

import DragVehicle from "./DragVehicle";

const item = { id: 23, x: 23, y: 23, sort: 0 };

test("should render DragVehicle component", () => {
  const settleMark = jest.fn();
  const updateCooridnates = jest.fn();

  const { container } = render(
    <DragVehicle
      item={item}
      settleMark={settleMark}
      updateCooridnates={updateCooridnates}
    />
  );

  expect(container).toMatchSnapshot();
});
