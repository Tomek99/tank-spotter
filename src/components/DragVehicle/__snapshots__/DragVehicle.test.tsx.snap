// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DragVehicle component 1`] = `
<div>
  <div
    class="DragVehicle"
    style="position: absolute; top: 0px; left: 0px;"
  >
    <div
      class="Artillery"
    />
  </div>
</div>
`;
